File: Dataset/Leetcode/train/110/222.java
class Solution {
    public boolean XXX(TreeNode root) {
        if(root == null){
            return true;
        }
        return XXX01(root) != -1;
    }

    public int XXX01(TreeNode root) {
        if(root == null){
            return 0;
        }

        int left = XXX01(root.left);
        int right = XXX01(root.right);

        if (left == -1 || right == -1){
            return -1;
        } else if (Math.abs(left - right) > 1){
            return -1;
        }

        return Math.max(left,right) + 1;

    }
}

File: Dataset/Leetcode/train/9/222.cpp
class Solution {
public:
    bool XXX(int x) {
        if(x < 0) return false;
        int div = 1;
        while(x / div >= 10) {
            div *= 10;
        }
        while(x > 0) {
            int f, l;
            l = x % 10;
            f = x / div;
            if(l != f)
                return false;
            x = x % div / 10;
            div = div / 100;
        }
        return true;
    }
};

